boot/bootutil/src/bootutil_misc.c

```diff
@@ -291,16 +300,14 @@ boot_read_swap_state_by_id(int flash_area_id, struct boot_swap_state *state)
     const struct flash_area *fap;
     int rc;
 
-    switch (flash_area_id) {
```

Comment:
Previous switch-case consumes 16 B less for for-size optimization using zephyr-sdk-0.10.1

Reply:
I would do similar analize for similar changes.

Reply:
The FLASH_AREA_IMAGE_* macros was changed from static #defines to dynamically (run-time) evaluated macros to hide which flash area belongs to a certain image primary and secondary slot. In a switch-case the expression must be constant, due to this reason the switch-case was transformed to the if () statement. I will take a look how can this be optimised for single image case.

Reply:
I see

Reply:
@davidvincze ^^